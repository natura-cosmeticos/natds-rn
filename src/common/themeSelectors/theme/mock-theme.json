{
  "avatarSizes": {
    "tiny": {
      "size": 24,
      "fontSize": ".625rem"
    },
    "small": {
      "size": 32,
      "fontSize": ".875rem"
    },
    "standard": {
      "size": 40,
      "fontSize": "1rem"
    },
    "large": {
      "size": 48,
      "fontSize": "1rem"
    },
    "huge": {
      "size": 80,
      "fontSize": "1.5rem"
    }
  },
  "buttonSizes": {
    "small": {
      "paddingTop": 8,
      "paddingRight": 8,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "minHeight": 32
    },
    "medium": {
      "paddingTop": 12,
      "paddingRight": 16,
      "paddingBottom": 12,
      "paddingLeft": 16,
      "minHeight": 40
    },
    "large": {
      "paddingTop": 16,
      "paddingRight": 16,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "minHeight": 48
    }
  },
  "fontSize": {
    "h1": {
      "fontSize": "6rem",
      "fontWeight": 300
    },
    "h2": {
      "fontSize": "3.75rem",
      "fontWeight": 300
    },
    "h3": {
      "fontSize": "3rem",
      "fontWeight": 400
    },
    "h4": {
      "fontSize": "2.125rem",
      "fontWeight": 400
    },
    "h5": {
      "fontSize": "1.5rem",
      "fontWeight": 400
    },
    "h6": {
      "fontSize": "1.25rem",
      "fontWeight": 600
    },
    "subtitle1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "subtitle2": {
      "fontSize": "0.875rem",
      "fontWeight": 500
    },
    "body1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "body2": {
      "fontSize": "0.875rem",
      "fontWeight": 400
    },
    "button": {
      "fontSize": "0.875rem",
      "fontWeight": 600
    },
    "caption": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    },
    "overline": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    }
  },
  "iconSizes": {
    "micro": 16,
    "tiny": 24,
    "small": 32,
    "standard": 48,
    "medium": 64,
    "large": 72,
    "largex": 96,
    "largexx": 128,
    "huge": 144,
    "hugex": 192,
    "hugexx": 256
  },
  "opacity": {
    "12": 0.12,
    "16": 0.16,
    "24": 0.24,
    "32": 0.32,
    "48": 0.48,
    "56": 0.56,
    "64": 0.64,
    "80": 0.8,
    "00": 0,
    "04": 0.04,
    "08": 0.08
  },
  "overlay": {
    "solidBlack": 0.56
  },
  "radius": {
    "none": 0,
    "circle": "50%",
    "small": 2,
    "medium": 4,
    "large": 8
  },
  "sizes": {
    "none": 0,
    "micro": 4,
    "tiny": 8,
    "small": 16,
    "standard": 24,
    "semi": 32,
    "semix": 40,
    "medium": 48,
    "mediumx": 56,
    "large": 64,
    "largex": 72,
    "largexx": 80,
    "largexxx": 88,
    "huge": 96,
    "hugex": 128,
    "hugexx": 144,
    "hugexxx": 192,
    "veryhuge": 256
  },
  "spacing": {
    "spacing": 8,
    "spacingNone": 0,
    "spacingMicro": 4,
    "spacingTiny": 8,
    "spacingSmall": 16,
    "spacingStandard": 24,
    "spacingSemi": 32,
    "spacingLarge": 48,
    "spacingHuge": 64
  },
  "colorTokens": {
    "colorPrimary": "#F4AB34",
    "colorOnPrimary": "#333333",
    "colorPrimaryLight": "#FEFDE8",
    "colorOnPrimaryLight": "#333333",
    "colorPrimaryDark": "#EF8426",
    "colorOnPrimaryDark": "#333333",
    "colorSecondary": "#FF6B0B",
    "colorOnSecondary": "#000000",
    "colorSecondaryLight": "#FFF8E1",
    "colorOnSecondaryLight": "#000000",
    "colorSecondaryDark": "#FF5808",
    "colorOnSecondaryDark": "#000000",
    "colorBackground": "#FAFAFA",
    "colorOnBackground": "#333333",
    "colorSurface": "#FFFFFF",
    "colorOnSurface": "#333333",
    "colorHighlight": "#000000",
    "colorHighEmphasis": "#333333",
    "colorMediumEmphasis": "#777777",
    "colorLowEmphasis": "#BBBBBB",
    "colorLink": "#227BBD",
    "colorOnLink": "#FFFFFF",
    "colorSuccess": "#569A32",
    "colorOnSuccess": "#FFFFFF",
    "colorWarning": "#FCC433",
    "colorOnWarning": "#333333",
    "colorAlert": "#E74627",
    "colorOnAlert": "#FFFFFF"
  },
  "elevation": {
    "1": {
      "elevation": 1,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 1, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 1
    },
    "2": {
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 2, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 2
    },
    "3": {
      "elevation": 3,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 3, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 4
    },
    "4": {
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 4, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 5
    },
    "5": {
      "elevation": 5,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 6, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "6": {
      "elevation": 6,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 8, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "7": {
      "elevation": 7,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 9, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 12
    },
    "8": {
      "elevation": 8,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 12, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 17
    },
    "9": {
      "elevation": 9,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 16, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 24
    },
    "10": {
      "elevation": 10,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 24, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 38
    }
  }
}
