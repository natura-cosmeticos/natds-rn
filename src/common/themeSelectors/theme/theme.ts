import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
