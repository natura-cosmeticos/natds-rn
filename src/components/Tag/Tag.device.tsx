import { storiesOf } from '@storybook/react-native';
import {
  Default, Borders, Sizes, Colors, Interactive,
} from './Tag.stories';

storiesOf('Tag', module)
  .add('Default', Default)
  .add('Borders', Borders)
  .add('Sizes', Sizes)
  .add('Colors', Colors)
  .add('Interactive', Interactive);
