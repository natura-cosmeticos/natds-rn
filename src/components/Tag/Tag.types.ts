/* eslint-disable max-len */
import { Color, Size } from '@naturacosmeticos/natds-themes/react-native'
import { AccessibilityRole, AccessibilityState } from 'react-native'

export type TagColors = keyof Pick<Color, 'primary' | 'secondary' | 'success' | 'alert' | 'warning' | 'link'>
export type TagSizes = keyof Pick<Size, 'small' | 'standard'>
// It was done that way for the documentation to show types explicitly.
export type TagPositions = keyof {
  default: 'default';
  left: 'left';
  right: 'right';
}

export interface TagProps {
  /**
   * Indicates that the view is an accessibility element.
   * @optional
   */
  accessible?: boolean;
  /**
   * When a view is marked as accessible, it is a good practice to set an accessibilityLabel on the view.
   * @optional
   */
  accessibilityLabel?: string;
  /**
   * An accessibility hint helps users understand what will happen when they
   * perform an action on the accessibility element when that result is not clear from the accessibility label.
   * @optional
   */
  accessibilityHint?: string;
  /**
   * Communicates the purpose of a component to the user of an assistive technology.
   * @optional
   */
  accessibilityRole?: AccessibilityRole;
  /**
   * Tells the screen reader to treat the currently focused on element as being in a specific state.
   * @example {selected: true, disabled: true}.
   * @optional
   */
  accessibilityState?: AccessibilityState;
  /** Specifies whether fonts should scale to respect Text Size accessibility settings.
   */
  allowFontScaling?: boolean;
  /** Specifies the largest possible scale a font can reach when allowFontScaling is enabled.
   * @default `undefined`
   */
  maxFontSizeMultiplier?: number;
  /** Define a label for the tag.
   * @required
   */
  text: string;
  /** Optional position of the tag.
   * @default `default`
   * @optional
   */
  borderPosition?: TagPositions;
  /** Optional size of the tag.
   * @default `small`
   * @optional
   */
  size?: TagSizes;
  /**
   * Optional color of the tag.
   * @default `primary`
   * @optional
   */
  color?: TagColors;
  /**
   * Optional ID for testing.
   * @optional
   */
  testID?: string;
}
