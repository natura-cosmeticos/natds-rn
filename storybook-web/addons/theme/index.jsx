import React, { useEffect, Fragment } from 'react';
import { useAddonState, Consumer } from '@storybook/api';
import { TabButton, WithTooltip, TooltipLinkList } from '@storybook/components';
import {
  PANEL_ID, CHANGE, PARAM_KEY, getStoryBookTheme,
} from './shared';
import { light, dark } from '../../theme';

import './styles.css';

const DEFAULT_THEME = {
  name: 'natura',
  type: 'light',
};

const buildThemeProps = mode => ({ ...DEFAULT_THEME, type: mode });

function parseTheme(themes) {
  return Object.entries(themes).reduce((accum, [name, variants]) => {
    Object.entries(variants).forEach(([type]) => {
      accum.push({ name, type });
    });

    return accum;
  }, []);
}

const createBackgroundSelectorItem = ({ name, type }, onChange, selectedItem) => ({
  active: selectedItem.name === name && selectedItem.type === type,
  id: `${name} - ${type}`,
  onClick: () => onChange({ name, type }),
  title: `${name} - ${type}`,
});

const getDisplayedItems = (list, onChange, selectedItem) => {
  if (!list.length) {
    return [];
  }

  return [
    ...list.map(params => createBackgroundSelectorItem(params, onChange, selectedItem)),
  ];
};

const mapper = ({ api, state }) => {
  const story = state.storiesHash[state.storyId];
  const { themes = {} } = (story && api.getParameters(story.id, PARAM_KEY)) || {};

  return { items: parseTheme(themes) };
};

export default function Theme({ channel, api }) {
  const apiTheme = getStoryBookTheme();
  const initialTheme = apiTheme ? buildThemeProps(apiTheme) : DEFAULT_THEME;
  const [currentTheme, changeTheme] = useAddonState(PANEL_ID, initialTheme);
  const { disabled } = api.getCurrentParameter(PARAM_KEY) || {};

  const handleChange = (params) => {
    changeTheme(params);
    api.setOptions({
      theme: params.type === 'light' ? light : dark,
    });
    const event = new Event('CHANGE_THEME', { detail: params.type });

    window.dispatchEvent(event);
    channel.emit(CHANGE, params);
  };

  const renderTooltip = items => ({ onHide }) => (
    <TooltipLinkList
      links={getDisplayedItems(items, (index) => {
        handleChange(index);
        onHide();
      }, currentTheme)}
    />
  );

  const eventListener = event => handleChange(buildThemeProps(event.matches ? 'dark' : 'light'));

  useEffect(() => {
    handleChange(initialTheme);

    const matchMediaEnvent = window.matchMedia('(prefers-color-scheme: dark)');

    matchMediaEnvent.addEventListener('change', eventListener);

    return () => {
      matchMediaEnvent.removeEventListener('change', eventListener);
    };
  }, []);

  if (disabled) return null;

  return (
    <Consumer filter={mapper}>
      {({ items }) => (items.length ? (
          <Fragment>
            <WithTooltip
              placement="top"
              trigger="click"
              closeOnClick
              tooltip={renderTooltip(items)}
            >
              <TabButton
                key="Theme"
                title="Change the Theme">
                {currentTheme.name} - {currentTheme.type}
              </TabButton>
            </WithTooltip>
          </Fragment>
      ) : null)}
    </Consumer>
  );
}
