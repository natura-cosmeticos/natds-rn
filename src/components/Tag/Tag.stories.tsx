/* eslint-disable max-lines */
import { select, text } from '@storybook/addon-knobs';
import React from 'react';

import { StoryContainer } from '../../common/HelperComponents/StoryContainer';
import { Tag } from './Tag';
import { TagColors, TagPositions, TagSizes } from './Tag.types';

const description = () => `
- - -

  ### NOTE:
  This component is available in the following variants:

  - ✅ Standard

With the following attribute status:

  - **Attributes:**
    - **Size:**
      - ✅ \`Standard\`
      - ✅ \`Small\`
    - **Color:**
      - ✅ \`Primary\`
      - ✅ \`Secondary\`
      - ✅ \`Alert\`
      - ✅ \`Success\`
      - ✅ \`Link\`
      - ✅ \`Warining\`
    - **Position:**
      - ✅ \`Center\`
      - ✅ \`Left\`
      - ✅ \`Right\`

- - -
`;

const tagColors = [
  'alert',
  'link',
  'primary',
  'secondary',
  'success',
  'warning',
];

export default {
  component: Tag,
  parameters: {
    componentSubtitle: 'Tags are used to label, categorize, or organize items using keywords that describe them.',
    docs: {
      extractComponentDescription: description,
    },
  },
  title: 'Components|Tag',
};

export const Default = () => (
  <StoryContainer title='Standard'>
    <Tag text='Design System' />
  </StoryContainer>
);

export const Borders = () => (
  <>
    <StoryContainer title='Default'>
      <Tag text='Design System' borderPosition='default' />
    </StoryContainer>
    <StoryContainer title='Right'>
      <Tag text='Design System' borderPosition='right' />
    </StoryContainer>
    <StoryContainer title='Left'>
      <Tag text='Design System' borderPosition='left' />
    </StoryContainer>
  </>
);

export const Sizes = () => (
  <>
    <StoryContainer title='Standard'>
      <Tag text='Design System' size="standard" />
    </StoryContainer>
    <StoryContainer title='Small'>
      <Tag text='Design System' size="small" />
    </StoryContainer>
  </>
);

export const Colors = () => (
  <>
    <StoryContainer title='Primary'>
      <Tag text='Design System' color='primary' />
    </StoryContainer>
    <StoryContainer title='Secondary'>
      <Tag text='Design System' color='secondary' />
    </StoryContainer>
    <StoryContainer title='Alert'>
      <Tag text='Design System' color='alert' />
    </StoryContainer>
    <StoryContainer title='Warning'>
      <Tag text='Design System' color='warning' />
    </StoryContainer>
    <StoryContainer title='Success'>
      <Tag text='Design System' color='success' />
    </StoryContainer>
    <StoryContainer title='Link'>
      <Tag text='Design System' color='link' />
    </StoryContainer>
  </>
);

export const Interactive = () => (
  <StoryContainer title="Interactive">
    <Tag
      text={text('Text', 'Design System')}
      size={select('Size', ['small', 'standard'], 'small') as TagSizes}
      color={select('Color', tagColors, 'primary') as TagColors}
      borderPosition={select('Border Position', ['default', 'left', 'right'], 'default') as TagPositions}
    />
  </StoryContainer>
);
