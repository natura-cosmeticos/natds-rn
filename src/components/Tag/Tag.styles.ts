import styled, { css } from 'styled-components/native';
import { Theme } from '@naturacosmeticos/natds-themes/react-native';
import { TagColors, TagPositions, TagSizes } from './Tag.types';

type TagStyleBaseProps = {
  theme: Theme
}

type ContainerStyleProps = TagStyleBaseProps & {
  color?: TagColors
  size?: TagSizes
  borderPosition?: TagPositions
}

type LabelStyleProps = TagStyleBaseProps & {
  color?: TagColors
}

export const getBorderPosition = ({ theme, size = 'small', borderPosition = 'default' } : ContainerStyleProps) => {
  const styles = {
    default: {
      borderRadius: theme.tag[size].borderRadius.enable,
    },
    left: {
      borderBottomLeftRadius: theme.tag[size].borderRadius.enable,
      borderBottomRightRadius: theme.tag[size].borderRadius.disable,
      borderTopLeftRadius: theme.tag[size].borderRadius.enable,
      borderTopRightRadius: theme.tag[size].borderRadius.disable,
    },
    right: {
      borderBottomLeftRadius: theme.tag[size].borderRadius.disable,
      borderBottomRightRadius: theme.tag[size].borderRadius.enable,
      borderTopLeftRadius: theme.tag[size].borderRadius.disable,
      borderTopRightRadius: theme.tag[size].borderRadius.enable,
    },
  };

  return styles[borderPosition];
};

export const getTextColorByName = ({ theme, color = 'primary' } : LabelStyleProps) => {
  const styles = {
    alert: theme.color.onAlert,
    link: theme.color.onLink,
    primary: theme.color.onPrimary,
    secondary: theme.color.onSecondary,
    success: theme.color.onSuccess,
    warning: theme.color.onWarning,
  };

  return styles[color];
};

export const getPaddingBySize = ({ theme, size = 'small' } : ContainerStyleProps) => {
  if (size === 'standard') {
    return {
      paddingHorizontal: theme.spacing.tiny,
      paddingVertical: theme.spacing.micro,
    };
  }

  return { paddingHorizontal: theme.spacing.tiny };
};

export const Container = styled.View<ContainerStyleProps>(({
  theme, color = 'primary', size, borderPosition,
}: ContainerStyleProps) => ({
  ...getBorderPosition({ borderPosition, size, theme }),
  ...getPaddingBySize({ size, theme }),
  alignSelf: 'flex-start',
  backgroundColor: theme.color[color],
}));

export const Label = styled.Text<LabelStyleProps>`
  ${({ theme, color }: LabelStyleProps) => css`
    color: ${getTextColorByName({ color, theme })};
    font-family: ${theme.tag.label.primary.fontFamily};
    font-size: ${theme.tag.label.fontSize}px;
    font-weight: ${theme.tag.label.primary.fontWeight};
    letter-spacing: ${theme.tag.label.letterSpacing}px;
    line-height: ${theme.tag.label.fontSize * theme.tag.label.lineHeight}px;
  `}
`;
