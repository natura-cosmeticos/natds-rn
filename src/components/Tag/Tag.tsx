import React from 'react';

import { Container, Label } from './Tag.styles';
import { TagProps } from './Tag.types';

export const Tag = (props: TagProps) => {
  const {
    testID = 'ds-tag', color = 'primary', size = 'small', borderPosition = 'default',
  } = props;

  return (
    <Container
      testID={testID}
      color={color}
      size={size}
      borderPosition={borderPosition}
    >
      <Label
        color={color}
        testID={`${testID}-label`}
        accessible={props.accessible}
        accessibilityRole={props.accessibilityRole}
        accessibilityHint={props.accessibilityHint}
        accessibilityLabel={props.accessibilityLabel}
        accessibilityState={props.accessibilityState}
        allowFontScaling={props.allowFontScaling}
        maxFontSizeMultiplier={props.maxFontSizeMultiplier}
      >
        {props.text}
      </Label>
    </Container>
  );
};
