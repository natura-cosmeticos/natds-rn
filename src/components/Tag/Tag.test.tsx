import React from 'react';
import {
  getBorderPosition, getPaddingBySize, getTextColorByName,
} from './Tag.styles';

import { renderWithTheme } from '../../../test/testHelpers';
import { Tag } from './Tag';
import theme from '../../common/themeSelectors/theme/mock-theme.json';

const defaultProps = {
  text: 'Lorem Ipsum',
};

jest.mock('../../common/themeSelectors');

describe('Tag Component', () => {
  it('should render with default variant', () => {
    const { toJSON } = renderWithTheme(<Tag {...defaultProps} testID='tag'/>);

    expect(toJSON()).toMatchSnapshot();
  });
});

describe('Tag Colors', () => {
  it('should render with secondary color', () => {
    const { getByTestId, toJSON } = renderWithTheme(<Tag {...defaultProps} color='secondary'/>);

    expect(getByTestId('ds-tag')?.props).toHaveProperty('color', 'secondary');
    expect(getByTestId('ds-tag')).toHaveStyle({ backgroundColor: '#ff6b0b' });
    expect(getByTestId('ds-tag-label')).toHaveStyle({ color: '#000000' });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should return the standard color value', () => {
    const mockFn = jest.fn(getTextColorByName);

    expect(mockFn({ theme })).toEqual('#333333');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});

describe('Tag Sizes', () => {
  it('should render with non default size', () => {
    const { getByTestId, toJSON } = renderWithTheme(<Tag {...defaultProps} size='standard'/>);

    expect(getByTestId('ds-tag')?.props).toHaveProperty('size', 'standard');
    expect(getByTestId('ds-tag')).toHaveStyle({
      paddingHorizontal: 8,
      paddingVertical: 4,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should return default padding value ', () => {
    const mockFn = jest.fn(getPaddingBySize);

    expect(mockFn({ theme })).toEqual({ paddingHorizontal: 8 });
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});

describe('Tag Border Position', () => {
  it('should render with non standard border', () => {
    const { getByTestId, toJSON } = renderWithTheme(<Tag {...defaultProps} borderPosition='left'/>);

    expect(getByTestId('ds-tag')?.props).toHaveProperty('borderPosition', 'left');
    expect(getByTestId('ds-tag')).toHaveStyle({
      borderBottomLeftRadius: 8,
      borderTopLeftRadius: 8,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should return standard border value', () => {
    const mockFn = jest.fn(getBorderPosition);

    expect(mockFn({ theme })).toEqual({ borderRadius: 8 });
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
