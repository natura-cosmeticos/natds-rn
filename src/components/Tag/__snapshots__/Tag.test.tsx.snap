// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Border Position should render with non standard border 1`] = `
<View
  borderPosition="left"
  color="primary"
  size="small"
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#f4ab34",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 0,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 0,
        "paddingHorizontal": 8,
      },
    ]
  }
  testID="ds-tag"
>
  <Text
    color="primary"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.38,
          "lineHeight": 15,
        },
      ]
    }
    testID="ds-tag-label"
  >
    Lorem Ipsum
  </Text>
</View>
`;

exports[`Tag Colors should render with secondary color 1`] = `
<View
  borderPosition="default"
  color="secondary"
  size="small"
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#ff6b0b",
        "borderRadius": 8,
        "paddingHorizontal": 8,
      },
    ]
  }
  testID="ds-tag"
>
  <Text
    color="secondary"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.38,
          "lineHeight": 15,
        },
      ]
    }
    testID="ds-tag-label"
  >
    Lorem Ipsum
  </Text>
</View>
`;

exports[`Tag Component should render with default variant 1`] = `
<View
  borderPosition="default"
  color="primary"
  size="small"
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#f4ab34",
        "borderRadius": 8,
        "paddingHorizontal": 8,
      },
    ]
  }
  testID="tag"
>
  <Text
    color="primary"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.38,
          "lineHeight": 15,
        },
      ]
    }
    testID="tag-label"
  >
    Lorem Ipsum
  </Text>
</View>
`;

exports[`Tag Sizes should render with non default size 1`] = `
<View
  borderPosition="default"
  color="primary"
  size="standard"
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#f4ab34",
        "borderRadius": 12,
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
    ]
  }
  testID="ds-tag"
>
  <Text
    color="primary"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.38,
          "lineHeight": 15,
        },
      ]
    }
    testID="ds-tag-label"
  >
    Lorem Ipsum
  </Text>
</View>
`;
